lib/amqp/Publication.js

```diff
@@ -46,13 +46,17 @@ function Publication(getChannelFn, publishFn, vhost, config) {
 
     function _publish(buffer, publishConfig, next) {
         var emitter = new EventEmitter()
-        getChannelFn(vhost, emitter, config, function(err, channel) {
-            if (err) return emitter.emit('error', err)
-            channel.on('return', emitter.emit.bind(emitter, 'return'))
+        var errorHandler = handleChannelError.bind(null, emitter, config)
+        var returnHandler = emitter.emit.bind(emitter, 'return')
+        borrowChannelFn(vhost, errorHandler, returnHandler, function(err, channel) {
+            if (err) {
+                if (channel) returnChannelFn(vhost, channel, errorHandler, returnHandler)
+                return emitter.emit('error', err)
+            }
             publishFn(channel, buffer, publishConfig, function(err) {
-                if (err) return emitter.emit('error', err)  // Channel will already be closed, reclosing will trigger an error
-                channel.close()
-                emitter.emit('success', publishConfig.options.messageId)
+              returnChannelFn(vhost, channel, errorHandler, returnHandler)
```

Comment:
We need to return channels to the pool, otherwise we'll starve them. 

Also on the topic of publish errors. If I understood amqplib code correctly, the only error this callback will get is this [nack](https://github.com/squaremo/amqp.node/blob/31bb2a9c40219ce2fc8c4fede15a84a9f54b36a1/lib/channel.js#L34). Which doesn't really signal that channel itself errored, so I think channel might actually be OK for reuse. And we also have a [handler for channel error events](https://github.com/BorePlusPlus/rascal/blob/c37fc5704e7ce228411091937ffb8cd7afaa2650/lib/amqp/Vhost.js#L114), which will mark channel as closed on error, so it won't get reused.